### PR TITLE
Add ScalaPB integration

### PR DESCRIPTION
**Pull request by DavidGregory084**
Adds a `ScalaPBModule` which configures a `ScalaModule` so that the [protobuf](https://developers.google.com/protocol-buffers/) compiler is used to generate Scala sources from .proto files in the `protobuf` folder in the module using [ScalaPBC](https://scalapb.github.io/scalapbc.html).

I used the Twirl integration as a guide to implement this, so there may be some things I am doing that are not necessary (e.g. calling ScalaPBC via reflection instead of as a subprocess). Let me know if I need to change that.

I added an example to the scratch module, which can be tested with`mill -i dev.run scratch scalapb.compile`

### CODE REVIEW

File: scalapblib/test/src/mill/scalapblib/TutorialTests.scala [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+          evalCount > 0
+        )
+
+        // don't recompile if nothing changed
+        val Right((_, unchangedEvalCount)) = eval.apply(Tutorial.core.compileScalaPB)
+
+        assert(unchangedEvalCount == 0)
+      }
+
+      // This throws a NullPointerException in coursier somewhere
```
Comment (DavidGregory084) [inline scalapblib/test/src/mill/scalapblib/TutorialTests.scala]: For some reason calling `compile` on this module causes an NPE in coursier somewhere. I had some trouble debugging mill via remote debug - breakpoints never get hit - so I haven't got to the bottom of this one yet.

The top of the stack trace looks like this:

```
[35/36] core.upstreamCompileOutput
java.lang.NullPointerException
	at coursier.core.Parse$.versionInterval(Parse.scala:60)
	at coursier.ivy.IvyRepository.find(IvyRepository.scala:156)
	at coursier.Fetch$.$anonfun$find$1(Fetch.scala:40)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
	at scala.collection.immutable.List.map(List.scala:295)
	at coursier.Fetch$.find(Fetch.scala:40)
	at coursier.Fetch$.get$1(Fetch.scala:68)
	at coursier.Fetch$.$anonfun$from$2(Fetch.scala:69)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:234)
	at scala.collection.Iterator.foreach(Iterator.scala:944)
	at scala.collection.Iterator.foreach$(Iterator.scala:944)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1432)
	at scala.collection.IterableLike.foreach(IterableLike.scala:71)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.collection.TraversableLike.map(TraversableLike.scala:234)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:227)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at coursier.Fetch$.$anonfun$from$1(Fetch.scala:66)
	at coursier.core.ResolutionProcess$.$anonfun$fetchAll$11(ResolutionProcess.scala:198)
	at scalaz.concurrent.Task.$anonfun$flatMap$2(Task.scala:34)
	at scalaz.concurrent.Task$.Try(Task.scala:456)
	at scalaz.concurrent.Task.$anonfun$flatMap$1(Task.scala:34)
	at scala.Function1.$anonfun$andThen$1(Function1.scala:52)
	at scalaz.concurrent.Future.$anonfun$flatMap$1(Future.scala:58)
	at scalaz.concurrent.Future.step(Future.scala:118)
	at scalaz.concurrent.Future.unsafePerformListen(Future.scala:75)
	at scalaz.concurrent.Future.unsafePerformAsync(Future.scala:159)
	at scalaz.concurrent.Future.unsafePerformSync(Future.scala:184)
	at scalaz.concurrent.Task.unsafePerformSync(Task.scala:98)
	at mill.modules.Jvm$.resolveDependenciesMetadata(Jvm.scala:463)
	at mill.modules.Jvm$.resolveDependencies(Jvm.scala:397)
	at mill.modules.Util$.millProjectModule(Util.scala:70)
	at mill.scalalib.ScalaWorkerModule.$anonfun$classpath$2(ScalaWorkerApi.scala:25)
```

**Comment by lihaoyi** [comment]
@DavidGregory084 could you put this in a `contrib` folder? As discussed with others earlier, I'd like to do so to keep the repo neat and clearly delimit the "core" Mill modules from these more specific extensions

**Comment by DavidGregory084** [comment]
Yep, no problem

**Comment by DavidGregory084** [comment]
```
-------------- Running Tests mill.integration.local.CaffeineTests --------------
X mill.integration.local.CaffeineTests.test 118623ms 
  utest.AssertionError: assert(eval("guava.test"))
    utest.asserts.Asserts$.assertImpl(Asserts.scala:114)
    mill.integration.CaffeineTests.$anonfun$tests$2(CaffeineTests.scala:25)
1 targets failed
```
I don't think this is related to my change. @lihaoyi would you mind giving the travis build a kick?

I will see if I can figure out why the test is flaky.

**Comment by lihaoyi** [comment]
yeah that guy is flaky, kicking it

**Comment by DavidGregory084** [comment]
Looks like travis died again...although all of the CI jobs passed

**Comment by lihaoyi** [comment]
looks good to me

**Comment by DavidGregory084** [comment]
Thanks @lihaoyi!

**Pull request merged**